service/history/timer_queue_standby_task_executor.go

```diff
@@ -459,6 +459,11 @@ func (t *timerQueueStandbyTaskExecutor) processTimer(
 		// workflow already finished, no need to process the timer
 		return nil
 	}
+	nsRecord := mutableState.GetNamespaceEntry()
+	if !nsRecord.IsOnCluster(t.clusterName) {
+		// discard standby tasks
+		return nil
+	}
```

Comment:
nit: I still think this logic should live in `getStandbyPostActionFn`. `processTimer/Transfer` can be used for any task logic, so any new task might get dropped accidentally. But `getStandbyPostActionFn` is specifically for deciding what to do if a standby task can't be processed, and when invoking this function, developer is aware of the fact that the discard function he/she provided will be used. 

Although I understanding changes here will prevent task from being executed at all. So I don't have strong opinion.

Reply:
Yes. I try to prevent task being executed compare execute once and discard.